src/dash-table/components/ControlledTable/index.tsx

```diff
@@ -735,13 +717,30 @@ export default class ControlledTable extends PureComponent<ControlledTableProps>
         ];
 
         const rawTable = this.tableFn();
-        const grid = derivedTableFragments(
+        const { grid, empty } = derivedTableFragments(
             n_fixed_columns,
             n_fixed_rows,
             rawTable,
             virtualized.offset.rows
         );
 
+        const classes = [
+            'dash-spreadsheet',
+            ...(virtualization ? ['dash-virtualized'] : []),
+            ...(n_fixed_rows ? ['dash-freeze-top'] : []),
+            ...(n_fixed_columns ? ['dash-freeze-left'] : []),
+            ...(style_as_list_view ? ['dash-list-view'] : []),
+            ...(empty[0][1] ? ['dash-empty-01'] : []),
```

Comment:
Adding a few marker classes for various cases
- empty content & fixed contents
- no data, no columns, no filter